server/src/database/commands/clean-inactive-workspaces.command.ts

```diff
@@ -0,0 +1,237 @@
+import {
+  Command,
+  CommandRunner,
+  InquirerService,
+  Option,
+} from 'nest-commander';
+
+import { PrismaService } from 'src/database/prisma.service';
+import peopleSeed from 'src/core/person/seed-data/people.json';
+import companiesSeed from 'src/core/company/seed-data/companies.json';
+import pipelineStagesSeed from 'src/core/pipeline/seed-data/pipeline-stages.json';
+import pipelinesSeed from 'src/core/pipeline/seed-data/sales-pipeline.json';
+import { arraysEqual } from 'src/utils/equal';
+import { WorkspaceService } from 'src/core/workspace/services/workspace.service';
+
+interface DataCleanInactiveOptions {
+  days?: number;
+  sameAsSeedDays?: number;
+  dryRun?: boolean;
+  confirmation?: boolean;
+}
+
+interface ActivityReport {
+  displayName: string;
+  maxUpdatedAt: string;
+  inactiveDays: number;
+}
+
+interface SameAsSeedWorkspace {
+  displayName: string;
+}
+
+interface DataCleanResults {
+  activityReport: { [key: string]: ActivityReport };
+  sameAsSeedWorkspaces: { [key: string]: SameAsSeedWorkspace };
+}
+
+@Command({
+  name: 'workspaces:clean-inactive',
+  description: 'Clean inactive workspaces from the public database schema',
+})
+export class DataCleanInactiveCommand extends CommandRunner {
+  constructor(
+    private readonly prismaService: PrismaService,
+    private readonly workspaceService: WorkspaceService,
+    private readonly inquiererService: InquirerService,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-d, --days [inactive days threshold]',
+    description: 'Inactive days threshold',
+    defaultValue: 60,
+  })
+  parseDays(val: string): number {
+    return Number(val);
+  }
+
+  @Option({
+    flags: '-s, --same-as-seed-days [same as seed days threshold]',
+    description: 'Same as seed days threshold',
+    defaultValue: 10,
+  })
+  parseSameAsSeedDays(val: string): number {
+    return Number(val);
+  }
+
+  @Option({
+    flags: '--dry-run [dry run]',
+    description: 'List inactive workspaces without removing them',
+  })
+  parseDryRun(val: string): boolean {
+    return Boolean(val);
+  }
+
+  // We look for public tables which contains workspaceId and updatedAt columns
+  getRelevantTables() {
+    return Object.keys(this.prismaService.client).filter(
+      (name) =>
+        !name.startsWith('_') &&
+        !name.startsWith('$') &&
+        !name.includes('user') &&
+        !name.includes('refreshToken') &&
+        !name.includes('workspace'),
+    );
+  }
+
+  async getTableMaxUpdatedAt(table, workspace) {
+    try {
+      return await this.prismaService.client[table].aggregate({
+        _max: { updatedAt: true },
+        where: { workspaceId: { equals: workspace.id } },
+      });
+    } catch (e) {}
```

Comment:
why do we try catch and fail silently?

Reply:
in case no updatedAt column exists -> eg: favorites

Reply:
I remove favorites from the tables to check